File: task3.rb
# Task 3

# Put your code here

#Реализовать функцию расчета наибольшего общего делителя для чисел a и b.
#На вход подается два числа a и b, разделенные пробелом. 1 <= a, b <= 10000

#a = ARGV[0].to_i
#b = ARGV[1].to_i

def method_nod(a,b)
	nod = [a,b].min

	nod+=1
	begin
	  nod-=1
	  adiv = a%nod
	  bdiv = b%nod
	end until ((adiv+bdiv) == 0)
	return nod
end

puts method_nod(ARGV[0].to_i,ARGV[1].to_i)

File: task3.js
/* Task 3 */

/* Put your code here */

/*
Реализовать функцию расчета наибольшего общего делителя для чисел a и b.
На вход подается два числа a и b, разделенные пробелом. 1 <= a, b <= 10000

Пример
> node task.rb 10 8
> 2
*/

function fnod(_a,_b)
{
	var nod = Math.min(_a,_b);

	for(;nod>1;nod--)
		{
			var adiv = _a%nod;
			var bdiv = _b%nod;
			if(((adiv+bdiv) == 0)) 	break;
		}
	return nod;
}
//GB
process.stdout.write(fnod(process.argv[1],process.argv[2]).toString());

//me
//process.stdout.write(fnod(process.argv[2],process.argv[3]).toString());
